integrations/instagram/src/index.ts

```diff
@@ -260,33 +268,56 @@ export function getRecipientId(conversation: SendMessageProps['conversation']):
   return recipientId
 }
 
-async function handleMessage(message: InstagramMessage, client: bp.Client) {
-  if (message.message) {
-    if (message.message.text) {
-      const { conversation } = await client.getOrCreateConversation({
-        channel: 'channel',
-        tags: {
-          [idTag]: message.sender.id,
-        },
-      })
-
-      const { user } = await client.getOrCreateUser({
-        tags: {
-          [idTag]: message.sender.id,
-        },
-      })
-
-      await client.createMessage({
-        type: 'text',
-        tags: {
-          // TODO: declare in definition
-          // [idTag]: message.message.mid
-        },
-        userId: user.id,
-        conversationId: conversation.id,
-        payload: { text: message.message.text },
-      })
+async function handleMessage(
+  message: InstagramMessage,
+  {
+    client,
+    ctx,
+    logger,
+  }: { client: bp.Client; ctx: IntegrationContext<bp.configuration.Configuration>; logger: IntegrationLogger }
+) {
+  if (message?.message?.text) {
+    const { conversation } = await client.getOrCreateConversation({
+      channel: 'channel',
+      tags: {
+        [idTag]: message.sender.id,
+      },
+    })
+
+    const { user } = await client.getOrCreateUser({
+      tags: {
+        [idTag]: message.sender.id,
+      },
+    })
+
+    if (!user.pictureUrl || !user.name) {
+      try {
+        const messengerClient = getMessengerClient(ctx.configuration)
+        const userProfile = await messengerClient.getUserProfile(message.sender.id, {
+          fields: ['id', 'name', 'profile_pic'],
+        })
+        const fieldsToUpdate = {
+          pictureUrl: userProfile?.profilePic,
+          name: userProfile?.name,
+        }
+        if (fieldsToUpdate.pictureUrl || fieldsToUpdate.name) {
+          await client.updateUser({ ...user, ...fieldsToUpdate })
```

Comment:
**thought**: Maybe we should add a `logger.forBot().info()` here (and in all integrations) so if this ever goes haywire (e.g. the channel returning an empty name or picture URL, which would cause this to run for every message), at least it's made evident in the bot/integration logs?

Reply:
makes sense, let me add just that

Reply:
just added logs